test/es-module/test-esm-resolve-type.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+// Flags: --expose-internals
+
+/**
+ * This test ensures defaultResolve returns the found module format in the
+ * return object in the form:
+ * { url: <url_value>, format: <'module'|'commonjs'|undefined> };
+ */
+
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+const fixtures = require('../common/fixtures');
+const path = require('path');
+const fs = require('fs');
+
+const assert = require('assert');
+const {
+  defaultResolve: resolve
+} = require('internal/modules/esm/resolve');
+
+/**
+ * ensure that resolving by full path does not return the format
+ * with the defaultResolver
+ */
+[ [ '/es-modules/package-type-module/index.js', undefined ],
+  [ '/es-modules/package-type-commonjs/index.js', undefined ],
+  [ '/es-modules/package-without-type/index.js', undefined ],
+  [ '/es-modules/package-without-pjson/index.js', undefined ],
+].forEach((testVariant) => {
+  const [ testScript, expectedType ] = testVariant;
+  const resolvedPath = path.resolve(fixtures.path(testScript));
+  const resolveResult = resolve(resolvedPath);
+  assert.strictEqual(resolveResult.format, expectedType);
+});
+
+/**
+ * create a test module and try to resolve it by module name.
+ * check the result is as expected
+ */
+
+(async () => {
+  [ [ 'test-module-mainjs', 'js', 'module', 'module'],
+    [ 'test-module-mainmjs', 'mjs', 'module', 'module'],
+    [ 'test-module-cjs', 'js', 'commonjs', 'commonjs'],
+    [ 'test-module-ne', 'js', undefined, undefined],
+  ].forEach((testVariant) => {
+    const [ moduleName,
+            moduleExtenstion,
+            moduleType,
+            expectedResolvedType ] = testVariant;
+    tmpdir.refresh();
+    const rel = (file) => path.join(process.cwd(), file);
```

Comment:
this should be using the tmpdir and not cwd

Reply:
done with https://github.com/nodejs/node/pull/40980/commits/59350655e1d29fee1c56e6c0e194f28aed49b88d